JustSaying/Messaging/MessageProcessingStrategies/Throttled.cs

```diff
@@ -18,10 +18,16 @@ public Throttled(int maxWorkers, IMessageMonitor messageMonitor)
             _semaphore = new SemaphoreSlim(maxWorkers, maxWorkers);
         }
 
-        public void StartWorker(Func<Task> action)
+        public void StartWorker(Func<Task> action, CancellationToken ct)
         {
             var messageProcessingTask = new Task<Task>(() => ReleaseOnCompleted(action));
-            _semaphore.Wait();
+            _semaphore.Wait(ct);
+
+            if (ct.IsCancellationRequested)
+            {
+                return;
```

Comment:
Before returning, be sure to call `_semaphore.Release()` to keep the count correct

Reply:
I've changed this to catch `OperationCanceledException` instead